packages/dataviews/src/dataforms-layouts/regular/index.tsx

```diff
@@ -2,13 +2,31 @@
  * WordPress dependencies
  */
 import { __experimentalVStack as VStack } from '@wordpress/components';
-import { useMemo } from '@wordpress/element';
+import { memo, useMemo } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { normalizeFields } from '../../normalize-fields';
-import type { DataFormProps, Field } from '../../types';
+import type { DataFormProps, Field, NormalizedField } from '../../types';
+
+type MemoizedFieldEditProps< Item > = {
+	data: Item;
+	field: NormalizedField< Item >;
+	onChange: Pick< DataFormProps< Item >, 'onChange' >[ 'onChange' ];
+};
+
+const MemoizedFieldEdit = memo(
+	( { field, data, onChange } ) => (
+		<field.Edit data={ data } field={ field } onChange={ onChange } />
```

Comment:
What if instead of passing `data` we pass `field.getValue( { item: data } )` and we just memoize `FieldEdit` simply.

Reply:
Thanks for the review! You raise a great point. I'm not entirely sure—could it be that the data object might be useful for more complex edit field components?

Reply:
> could it be that the data object might be useful for more complex edit field components?

If it's the case, the current memoization will break it because it won't update if something else changes.